CoreObjectModel/MutableMetadataModel/Copier.cs

```diff
@@ -3306,9 +3306,9 @@ private void SubstituteElements(Dictionary<ITypeReference, IEnumerable<ICustomAt
       Contract.Requires(!(assembly is Dummy));
       Contract.Ensures(Contract.Result<Assembly>() != null);
 
-      this.TraverseAndPopulateDefinitionCacheWithCopies.Traverse(assembly); //Does not traverse the types
-      foreach (var type in assembly.GetAllTypes())
-        this.TraverseAndPopulateDefinitionCacheWithCopies.Traverse(type);
+      this.TraverseAndPopulateDefinitionCacheWithCopies.Traverse(assembly); // Traverses the types, except for the <Module> type
+      var moduleType = new List<INamedTypeDefinition>(assembly.GetAllTypes())[0];
```

Comment:
something like

      INamedTypeDefinition moduleType = assembly.GetAllTypes().FirstOrDefault();
      if (moduleType != null)
        this.TraverseAndPopulateDefinitionCacheWithCopies.Traverse(moduleType);

although we might need to slightly reformulate this to work without Linq support. I'm not sure whether façade assemblies have a module type, I need to ask Morgan; one way or another, the null check is perhaps better to stay on the safe side.